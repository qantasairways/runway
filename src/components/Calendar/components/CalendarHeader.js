import React from 'react';
import { css } from 'emotion';
import PropTypes from 'prop-types';
import Planeicon from '../../../icons/PlaneIcon';
import Triangle from '../../../icons/Triangle';
import { colours, mq, airwaysFontFamily } from '../../../theme/airways';
import {
  CSS_PSEUDO_AFTER,
  CSS_SELECTOR_FOCUS,
  CSS_SELECTOR_HOVER
} from '../../../constants/css';

export function calendarHeatherCntStyle() {
  return css({
    label: 'runway-calendar-header-style_container',
    position: 'relative',
    width: '100%',
    backgroundColor: colours.darkerGrey,
    display: 'flex'
  });
}

const tabStyles = {
  position: 'relative',
  cursor: 'pointer',
  width: '50%',
  height: '64px',
  display: 'flex',
  alignItems: 'center',
  transition: '.4s',
  [CSS_SELECTOR_FOCUS]: {
    outline: 'none'
  },
  [mq.medium]: {
    height: '80px'
  },
  [mq.large]: {
    justifyContent: 'center'
  }
};

export function returnTabStyle({ isSelectingStartDate }) {
  return css({
    label: 'runway-return_tab',
    color: isSelectingStartDate ? colours.white : colours.darkerGrey,
    border: isSelectingStartDate ? 'none' : 'solid 5px #8de2e0',
    borderLeft: 'none',
    backgroundColor: isSelectingStartDate ? colours.darkerGrey : colours.white,
    textAlign: 'right',
    float: 'right',
    justifyContent: 'flex-end',
    padding: '0 15px',
    ...tabStyles,
    [CSS_SELECTOR_HOVER]: {
      backgroundColor: isSelectingStartDate ? '#3c3c3c' : colours.white
    }
  });
}

export function departTabStyles({ isSelectingStartDate }) {
  return css({
    label: 'runway-depart_tab',
    padding: '0 15px',
    border: isSelectingStartDate ? 'solid 5px #8de2e0' : 'solid 5px #323232',
    borderRight: '0',
    backgroundColor: isSelectingStartDate ? colours.white : colours.darkerGrey,
    color: isSelectingStartDate ? colours.darkerGrey : colours.white,
    boxSizing: 'border-box',
    justifyContent: 'space-between',
    textAlign: 'left',
    ...tabStyles,
    [CSS_SELECTOR_HOVER]: {
      backgroundColor: isSelectingStartDate ? colours.white : '#3c3c3c'
    },
    [CSS_PSEUDO_AFTER]: {
      content: "''",
      zIndex: '5',
      position: 'absolute',
      right: isSelectingStartDate ? '-14px' : '0',
      borderTop: '32px solid transparent',
      borderLeft: isSelectingStartDate ? '14px solid white' : '0',
      borderBottom: '32px solid transparent',
      borderRight: isSelectingStartDate ? '0' : '19px solid white',
      top: '-5px',
      [mq.medium]: {
        right: isSelectingStartDate ? '-17px' : '-1px',
        borderLeft: isSelectingStartDate ? '19px solid white' : '0',
        borderTop: '42px solid transparent',
        borderBottom: '42px solid transparent',
        top: '-7px'
      }
    }
  });
}

export function textStyles({ isPlaceholder }) {
  return css({
    label: 'runway_text_style',
    fontFamily: airwaysFontFamily.main,
    fontSize: '18px',
    lineHeight: isPlaceholder ? 1.11 : 1.25,
    fontWeight: 500,
    display: 'flex',
    flexDirection: 'column',
    [mq.medium]: {
      fontSize: '26px'
    },
    [mq.large]: {
      flexDirection: 'row'
    }
  });
}

export function planeSvgStyles({ isSelectingStartDate }) {
  return css({
    label: 'runway-slider_plane-svg',
    width: '32px',
    height: '32px',
    zIndex: '10',
    position: 'absolute',
    right: isSelectingStartDate ? '-5px' : '-26px',
    transform: isSelectingStartDate ? 'rotate(0deg)' : 'rotate(180deg)',
    [mq.medium]: {
      width: '40px',
      height: '40px',
      right: isSelectingStartDate ? '-3px' : '-34px'
    },
    [mq.small]: {
      width: '28px',
      height: '28px',
      right: isSelectingStartDate ? '-7px' : '-20px'
    }
  });
}

export function TriangleSvgStyles({ isSelectingStartDate }) {
  return css({
    label: 'runway-slider_triangle-svg',
    zIndex: '10',
    position: 'absolute',
    right: isSelectingStartDate ? '-49px' : '-11px',
    top: isSelectingStartDate ? '-5px' : '-7px',
    transform: isSelectingStartDate ? 'rotate(0)' : 'rotate(-180deg)',
    [mq.medium]: {
      height: '83px',
      right: isSelectingStartDate ? '-53px' : '-6px',
      top: isSelectingStartDate ? '-5px' : '-8px'
    }
  });
}

const CalendarHeader = ({
  isSelectingStartDate,
  departSelectedLabel,
  returnSelectedLabel,
  departPlaceholder,
  returnPlaceholder,
  departDate,
  returnDate,
  onTabClick,
  isOneWay
}) => {
  const renderDepartPlaceholder = () => {
    if (departDate) {
      return (
        <div css={textStyles({ isPlaceholder: false })}>
          <span>
            <b>
              {departSelectedLabel}
              &nbsp;
            </b>
          </span>
          <span>{departDate}</span>
        </div>
      );
    }
    return (
      <span css={textStyles({ isPlaceholder: true })}>{departPlaceholder}</span>
    );
  };
  const renderReturnPlaceholder = () => {
    if (returnDate) {
      return (
        <div css={textStyles({ isPlaceholder: false })}>
          <span>
            <b>{returnSelectedLabel}</b>
          </span>
          <span>
            &nbsp;
            {returnDate}
          </span>
        </div>
      );
    }
    return (
      <span css={textStyles({ isPlaceholder: true })}>{returnPlaceholder}</span>
    );
  };
  return (
    <div>
      <div css={calendarHeatherCntStyle()}>
        <button
          onClick={() => onTabClick({ isSelectingStartDate: true })}
          type="button"
          css={departTabStyles({ isSelectingStartDate })}
        >
          {renderDepartPlaceholder()}
          <Planeicon css={planeSvgStyles({ isSelectingStartDate })} />
          <Triangle css={TriangleSvgStyles({ isSelectingStartDate })} />
        </button>
        {!isOneWay && (
          <button
            onClick={() => onTabClick({ isSelectingStartDate: false })}
            type="button"
            css={returnTabStyle({ isSelectingStartDate })}
          >
            {renderReturnPlaceholder()}
          </button>
        )}
      </div>
    </div>
  );
};

CalendarHeader.propTypes = {
  isSelectingStartDate: PropTypes.bool,
  isOneWay: PropTypes.bool,
  departPlaceholder: PropTypes.string.isRequired,
  returnPlaceholder: PropTypes.string.isRequired,
  departSelectedLabel: PropTypes.string.isRequired,
  returnSelectedLabel: PropTypes.string.isRequired,
  departDate: PropTypes.string,
  returnDate: PropTypes.string,
  onTabClick: PropTypes.func
};

CalendarHeader.defaultProps = {
  isSelectingStartDate: true,
  isOneWay: false,
  departDate: null,
  returnDate: null,
  onTabClick: () => {}
};

export default CalendarHeader;
