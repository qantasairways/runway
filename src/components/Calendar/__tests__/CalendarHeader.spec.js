import React from 'react';
import { shallow } from 'enzyme';
import CalendarHeader from '../components/CalendarHeader';

describe('CalendarHeader', () => {
  it('renders with defaults', () => {
    const calendar = shallow(
      <CalendarHeader
        departPlaceholder="depart?"
        returnPlaceholder="return?"
        departSelectedLabel="departing on"
        returnSelectedLabel="returning on"
      />
    );

    expect(calendar).toMatchSnapshot();
  });

  it('renders with props provided', () => {
    const calendar = shallow(
      <CalendarHeader
        departPlaceholder="depart?"
        returnPlaceholder="return?"
        departSelectedLabel="departing on"
        returnSelectedLabel="returning on"
        isSelectingStartDate={false}
        departDate="6th June"
        returnDate="10th October"
      />
    );

    expect(calendar).toMatchSnapshot();
  });
});
