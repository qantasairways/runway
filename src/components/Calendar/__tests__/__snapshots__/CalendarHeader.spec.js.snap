// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader renders with defaults 1`] = `
.emotion-6 {
  position: relative;
  width: 100%;
  background-color: #323232;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  padding: 0 15px;
  border: solid 5px #8de2e0;
  border-right: 0px;
  background-color: #fff;
  color: #323232;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-align: left;
  position: relative;
  cursor: pointer;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: .4s;
  transition: .4s;
}

.emotion-3:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-3 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-3:hover {
  background-color: #fff;
}

.emotion-3::after {
  content: '';
  z-index: 5;
  position: absolute;
  right: -14px;
  border-top: 32px solid transparent;
  border-left: 14px solid white;
  border-bottom: 32px solid transparent;
  border-right: 0px;
  top: -5px;
}

@media only screen and (min-width:768px) {
  .emotion-3::after {
    right: -17px;
    border-left: 19px solid white;
    border-top: 42px solid transparent;
    border-bottom: 42px solid transparent;
    top: -7px;
  }
}

.emotion-0 {
  font-family: "Ciutadella",sans-serif;
  font-size: 18px;
  line-height: 1.11;
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 26px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-1 {
  width: 32px;
  height: 32px;
  z-index: 10;
  position: absolute;
  right: -5px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    width: 40px;
    height: 40px;
    right: -3px;
  }
}

@media only screen and (max-width:320px) {
  .emotion-1 {
    width: 28px;
    height: 28px;
    right: -7px;
  }
}

.emotion-2 {
  z-index: 10;
  position: absolute;
  right: -49px;
  top: -5px;
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    height: 83px;
    right: -53px;
    top: -5px;
  }
}

.emotion-5 {
  color: #fff;
  border: none;
  border-left: none;
  background-color: #323232;
  text-align: right;
  float: right;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0 15px;
  position: relative;
  cursor: pointer;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: .4s;
  transition: .4s;
}

.emotion-5:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-5 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-5 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-5:hover {
  background-color: #3c3c3c;
}

<div>
  <div
    className="emotion-6"
  >
    <button
      className="emotion-3"
      onClick={[Function]}
      type="button"
    >
      <span
        className="emotion-0"
      >
        depart?
      </span>
      <PlaneIcon
        className="emotion-1"
        height="24"
        width="24"
      />
      <Triangle
        className="emotion-2"
      />
    </button>
    <button
      className="emotion-5"
      onClick={[Function]}
      type="button"
    >
      <span
        className="emotion-0"
      >
        return?
      </span>
    </button>
  </div>
</div>
`;

exports[`CalendarHeader renders with props provided 1`] = `
.emotion-6 {
  position: relative;
  width: 100%;
  background-color: #323232;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  padding: 0 15px;
  border: solid 5px #323232;
  border-right: 0px;
  background-color: #323232;
  color: #fff;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-align: left;
  position: relative;
  cursor: pointer;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: .4s;
  transition: .4s;
}

.emotion-3:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-3 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-3:hover {
  background-color: #3c3c3c;
}

.emotion-3::after {
  content: '';
  z-index: 5;
  position: absolute;
  right: 0px;
  border-top: 32px solid transparent;
  border-left: 0px;
  border-bottom: 32px solid transparent;
  border-right: 19px solid white;
  top: -5px;
}

@media only screen and (min-width:768px) {
  .emotion-3::after {
    right: -1px;
    border-left: 0px;
    border-top: 42px solid transparent;
    border-bottom: 42px solid transparent;
    top: -7px;
  }
}

.emotion-0 {
  font-family: "Ciutadella",sans-serif;
  font-size: 18px;
  line-height: 1.25;
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 26px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-1 {
  width: 32px;
  height: 32px;
  z-index: 10;
  position: absolute;
  right: -26px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    width: 40px;
    height: 40px;
    right: -34px;
  }
}

@media only screen and (max-width:320px) {
  .emotion-1 {
    width: 28px;
    height: 28px;
    right: -20px;
  }
}

.emotion-2 {
  z-index: 10;
  position: absolute;
  right: -11px;
  top: -7px;
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    height: 83px;
    right: -6px;
    top: -8px;
  }
}

.emotion-5 {
  color: #323232;
  border: solid 5px #8de2e0;
  border-left: none;
  background-color: #fff;
  text-align: right;
  float: right;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0 15px;
  position: relative;
  cursor: pointer;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: .4s;
  transition: .4s;
}

.emotion-5:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-5 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-5 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-5:hover {
  background-color: #fff;
}

<div>
  <div
    className="emotion-6"
  >
    <button
      className="emotion-3"
      onClick={[Function]}
      type="button"
    >
      <div
        className="emotion-0"
      >
        <span>
          <b>
            departing on
             
          </b>
        </span>
        <span>
          6th June
        </span>
      </div>
      <PlaneIcon
        className="emotion-1"
        height="24"
        width="24"
      />
      <Triangle
        className="emotion-2"
      />
    </button>
    <button
      className="emotion-5"
      onClick={[Function]}
      type="button"
    >
      <div
        className="emotion-0"
      >
        <span>
          <b>
            returning on
          </b>
        </span>
        <span>
           
          10th October
        </span>
      </div>
    </button>
  </div>
</div>
`;
